lib/_tls_legacy.js

```diff
@@ -448,16 +448,16 @@ CryptoStream.prototype.destroy = function(err) {
   }
   this._opposite.destroy();
 
-  process.nextTick(destroyNT, this, err);
+  process.nextTick(destroyNT, this, err ? true : false);
 };
 
 
-function destroyNT(self, err) {
+function destroyNT(self, hadErr) {
   // Force EOF
   self.push(null);
 
   // Emit 'close' event
-  self.emit('close', err ? true : false);
+  self.emit('close', hadErr);
```

Comment:
Another reason: keeping an exception object alive for some time means keeping a complex object alive that in turn contains references to other objects.  A boolean is much (much!) easier for the GC to scan.